src/agr/vcf_file_generator.py

```diff
@@ -123,24 +103,23 @@ def __write_vcf_header(vcf_file, assembly, species, database_version):
            species=species,
            assembly=assembly)
         vcf_file.write(header)
-        vcf_file.write("\t".join(["#CHROM", "POS", "ID", "REF", "ALT", "QUAL", "FILTER", "INFO"]))
+        vcf_file.write('\t'.join(['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO']))
 
-    def __add_variant_to_vcf_file(vcf_file, variant):
-        if variant['hgvsNomenclature']:
-             hgvs = variant['hgvsNomenclature']
-        else:
-            hgvs = '.'
-        if variant['symbol']:
-            symbol = variant['symbol']
-        else:
-            symbol = '.'
-        if variant['alleleOfGenes']:
-            allele_of_genes = ', '.join(variant['alleleOfGenes'])
-        else:
-            allele_of_genes = '.'
-        info = 'hgvs_nomenclature="' + hgvs + '";symbol="' + symbol + '";allele_of_genes=' + allele_of_genes
+    @classmethod
+    def _variant_value_for_file(cls, variant, data_key, default='.'):
+        if variant[data_key]:
+            return variant[data_key]
+        return default
 
-        vcf_file.write("\n" + "\t".join([variant['chromosome'],
+    @classmethod
+    def _add_variant_to_vcf_file(cls, vcf_file, variant):
+        hgvs = cls._variant_value_for_file(variant, 'hgvsNomenclature')
+        if hgvs == '.':
+            info = '.'
+        else:
+            symbol = cls._variant_value_for_file(variant, 'symbol')
```

Comment:
It appears we are missing the allele_of_genes field. And we are missing the symbol being "." when it does not exist. the allele of genes should also be "." when does not exist (as far as I understand how VCF file INFO columns are supposed to work).

Reply:
INFO column logic should be correct for the current set of sub-fields in use.

Reply:
Added #7 - to be dealt with in separate PR.